File: mini-jdbc/src/main/java/com/mini/core/jdbc/MysqlJdbcTemplate.java
package com.mini.core.jdbc;

import javax.sql.DataSource;

import static java.lang.String.format;

public class MysqlJdbcTemplate extends JdbcTemplate {
    public MysqlJdbcTemplate() {
    }

    public MysqlJdbcTemplate(DataSource dataSource) {
        super(dataSource);
    }

    public MysqlJdbcTemplate(DataSource dataSource, boolean lazyInit) {
        super(dataSource, lazyInit);
    }

    @Override
    protected String paging(int start, int limit, String sql) {
        return format("%s LIMIT %d, %d", sql, start, limit);
    }

    @Override
    protected String totals(String sql) {
        return format("SELECT COUNT(*) FROM (%s) TB", sql);
    }
}

File: mini-jdbc/src/main/kotlin/com/mini/core/jdbc/MysqlJdbcTemplate.kt
@file:Suppress("unused")

package com.mini.core.jdbc

import javax.sql.DataSource

class MysqlJdbcTemplate @JvmOverloads constructor(
        dataSource: DataSource,
        lazyInit: Boolean = true
) : JdbcTemplate(dataSource, lazyInit) {
    override fun paging(start: Int, limit: Int, sql: String): String {
        return "$sql LIMIT $start, $limit"
    }

    override fun totals(sql: String): String {
        return "SELECT COUNT(*) FROM (${sql}) TB"
    }
}